app/controllers/spree/komoju_controller.rb

```diff
@@ -7,18 +7,44 @@ def callback
 
       case params[:type]
       when "payment.captured"
-        order_number = extract_payment_number(params[:data][:external_order_num])
-        payment = Spree::Payment.find_by_number!(order_number)
-        payment.complete! unless payment.completed?
+        payment_captured!
+      when "payment.refunded"
+        payment_refunded!
       end
 
       head 200
     end
 
     private
 
-    def extract_payment_number(external_order_num)
-      external_order_num.split('-').try(:last)
+    def order_number
+      params[:data][:external_order_num].split("-").try(:first)
+    end
+
+    def payment_number
+      params[:data][:external_order_num].split("-").try(:last)
+    end
+
+    def order
+      @order ||= Spree::Order.find_by_number!(order_number)
+    end
+
+    def payment
+      @payment ||= order.payments.find_by_number!(payment_number)
+    end
+
+    def payment_captured!
+      payment.complete! unless payment.completed?
+    end
+
+    def payment_refunded!
+      if payment.completed?
```

Comment:
@project-eutopia Can you add a test to check `completed?` returns false after a refund. I'm worried if this code doesn't work the app will go into an infinite loop between Spree and Komoju.